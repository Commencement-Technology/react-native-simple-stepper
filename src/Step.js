import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';

export default class Step extends Component {
  static propTypes = {
    activeOpacity: PropTypes.number,
    style: PropTypes.object,
    onPress: PropTypes.func,
    renderImage: PropTypes.func,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    activeOpacity: 1,
    style: {
      padding: 8,
    },
    onPress: () => {},
    renderImage: () => {
      return null;
    },
    disabled: false,
  };

  render() {
    const { activeOpacity, style, onPress, disabled, renderImage } = this.props;
    return (
      <TouchableOpacity activeOpacity={activeOpacity} style={style} onPress={onPress} disabled={disabled}>
        {renderImage()}
      </TouchableOpacity>
    );
  }
}
