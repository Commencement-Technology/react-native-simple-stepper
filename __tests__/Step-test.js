import React from 'react';
import Step from '../src/Step';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Step', () => {
  describe('renders', () => {
    test('snapshot', () => {
      const wrapper = shallow(<Step />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
    test('onPress', () => {
      const onPress = () => {};
      const wrapper = shallow(<Step />);
      wrapper.setProps({
        onPress,
      });
      expect(wrapper.prop('onPress')).toEqual(onPress);
      expect(wrapper.props().onPress).toBeDefined();
    });
  });
});
