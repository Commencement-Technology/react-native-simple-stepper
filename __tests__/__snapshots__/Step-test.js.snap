// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step renders snapshot 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={false}
  onPress={[Function]}
  style={
    Object {
      "padding": 8,
    }
  }
/>
`;
